src/main/java/org/spongepowered/api/event/cause/Cause.java

```diff
@@ -48,52 +52,211 @@
  * some blocks, but tracing this event would be too complicated and thus
  * may not be attempted.</p>
  */
-public class Cause {
+@SuppressWarnings("unchecked")
+public abstract class Cause {
```

Comment:
I'd rather that if you have an empty cause, it will never have to perform if checks for emptiness. I don't think it's necessary to rely on a `List` when `Cause` is final once created, there's no array instance for empty `Cause`s, there's no worry of anything. It's like `Optional` but for causes.